In [ ]:
from csv import reader
import pandas as pd
import matplotlib.pyplot as plt
import numpy as np
import seaborn as sns
import plotly.express as px
import country_converter as coco
!pip install pandoc

In [315]:
debt = pd.read_csv('international_debt.csv')

In [316]:
debt.head(5)

Country Country_code  \
0  Afghanistan          AFG   
1  Afghanistan          AFG   
2  Afghanistan          AFG   
3  Afghanistan          AFG   
4  Afghanistan          AFG   

                                      Indicator_name  Indicator_code  \
0  Disbursements on external debt, long-term (DIS...  DT.DIS.DLXF.CD   
1  Interest payments on external debt, long-term ...  DT.INT.DLXF.CD   
2                  PPG, bilateral (AMT, current US$)  DT.AMT.BLAT.CD   
3                  PPG, bilateral (DIS, current US$)  DT.DIS.BLAT.CD   
4                  PPG, bilateral (INT, current US$)  DT.INT.BLAT.CD   

         Debt  
0  72894453.7  
1  53239440.1  
2  61739336.9  
3  49114729.4  
4  39903620.1

In [317]:
custom_map = {
    'South Asia': 'Asia',
    'IDA only': 'Unspecified',
    'IDA': 'Unspecified',
    'DRC': 'Africa',
    'Least developed countries: UN classification': 'Unspecified'
}

# Manually define which countries are in South America
south_america_countries = {
    'Argentina', 'Bolivia', 'Brazil', 'Chile', 'Colombia', 'Ecuador',
    'Guyana', 'Paraguay', 'Peru', 'Suriname', 'Uruguay', 'Venezuela'
}

# Function to get continent
def get_continent(country):
    if country in custom_map:
        return custom_map[country]
    try:
        continent = coco.convert(names=country, to='continent')
        if continent == 'America':
            if country in south_america_countries:
                return 'South America'
            else:
                return 'North America'
        return continent
    except:
        return 'Unspecified'

# Apply function to dataframe
debt['Continent'] = debt['Country'].apply(get_continent)

In [318]:
debt.head(10)

Country Country_code  \
0  Afghanistan          AFG   
1  Afghanistan          AFG   
2  Afghanistan          AFG   
3  Afghanistan          AFG   
4  Afghanistan          AFG   
5  Afghanistan          AFG   
6  Afghanistan          AFG   
7  Afghanistan          AFG   
8  Afghanistan          AFG   
9  Afghanistan          AFG   

                                      Indicator_name  Indicator_code  \
0  Disbursements on external debt, long-term (DIS...  DT.DIS.DLXF.CD   
1  Interest payments on external debt, long-term ...  DT.INT.DLXF.CD   
2                  PPG, bilateral (AMT, current US$)  DT.AMT.BLAT.CD   
3                  PPG, bilateral (DIS, current US$)  DT.DIS.BLAT.CD   
4                  PPG, bilateral (INT, current US$)  DT.INT.BLAT.CD   
5               PPG, multilateral (AMT, current US$)  DT.AMT.MLAT.CD   
6               PPG, multilateral (DIS, current US$)  DT.DIS.MLAT.CD   
7               PPG, multilateral (INT, current US$)  DT.INT.MLAT.CD   
8         PPG, official creditors (AMT, current US$)  DT.AMT.OFFT.CD   
9         PPG, official creditors (DIS, current US$)  DT.DIS.OFFT.CD   

          Debt Continent  
0   72894453.7      Asia  
1   53239440.1      Asia  
2   61739336.9      Asia  
3   49114729.4      Asia  
4   39903620.1      Asia  
5   39107845.0      Asia  
6   23779724.3      Asia  
7   13335820.0      Asia  
8  100847181.9      Asia  
9   72894453.7      Asia

In [319]:
debt.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2357 entries, 0 to 2356
Data columns (total 6 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   Country         2357 non-null   object 
 1   Country_code    2357 non-null   object 
 2   Indicator_name  2357 non-null   object 
 3   Indicator_code  2357 non-null   object 
 4   Debt            2357 non-null   float64
 5   Continent       2357 non-null   object 
dtypes: float64(1), object(5)
memory usage: 110.6+ KB


In [320]:
debt.describe()

Debt
count  2.357000e+03
mean   1.306633e+09
std    5.226771e+09
min    0.000000e+00
25%    1.548248e+07
50%    1.073140e+08
75%    5.716575e+08
max    9.621862e+10

In [321]:
debt.duplicated()

0       False
1       False
2       False
3       False
4       False
        ...  
2352    False
2353    False
2354    False
2355    False
2356    False
Length: 2357, dtype: bool

# Debt Burden by Country

In [720]:
debtors = round(debt.groupby(['Country','Continent'])['Debt'].sum().sort_values(ascending=False),2).reset_index(name='Debt')

debtors.head(5)

Country      Continent          Debt
0                                         China           Asia  2.857935e+11
1                                        Brazil  South America  2.806240e+11
2                                    South Asia           Asia  2.476087e+11
3  Least developed countries: UN classification    Unspecified  2.128810e+11
4                            Russian Federation         Europe  1.912891e+11

`China` tops the world debtors list with `$285 billion`, closely followed by `Brazil` with `280 billion`

In [325]:
debtors = round(debt.groupby(['Country','Continent'])['Debt'].sum().sort_values(ascending=True),2).reset_index(name='Debt')

debtors.head(10)

Country      Continent         Debt
0  Sao Tome and Principe         Africa   44798032.5
1                  Tonga        Oceania   55209304.8
2                Comoros         Africa   57568483.2
3            Timor-Leste           Asia   71000959.2
4           Turkmenistan           Asia   95135629.3
5        Solomon Islands        Oceania  105104467.3
6                  Samoa        Oceania  112946425.5
7                Somalia         Africa  119614074.9
8                Eritrea         Africa  132493809.3
9               Dominica  North America  138044299.8

`Sao Tome and Principe` has the lowest debt with just `$44 million` in debt

# Debt Burden by Continent

In [328]:
debtors = round(debt.groupby(['Continent'])['Debt'].sum().sort_values(ascending=False),2).reset_index(name='Debt')

debtors.head(10)

Continent          Debt
0           Asia  1.318452e+12
1         Africa  4.395045e+11
2  South America  4.179970e+11
3    Unspecified  3.919291e+11
4         Europe  3.276597e+11
5  North America  1.745402e+11
6        Oceania  9.651777e+09

Asian countries appear to owe the most money with a total of `$1.3 trillion` in debt

# Average Debt Owed by Continent

In [331]:
avg_debt_by_cont = (debt.groupby(['Continent'])['Debt'].mean().reset_index().sort_values(by='Debt', ascending=False))
avg_debt_by_cont.head(25)

Continent          Debt
6    Unspecified  7.838582e+09
5  South America  2.402281e+09
1           Asia  2.079578e+09
2         Europe  1.274940e+09
3  North America  6.060425e+08
0         Africa  5.086858e+08
4        Oceania  1.072420e+08

`International Development Associations` owe the most amount on average when compared to other continents with `783 million`. It is also important to note that the countries under this class could actually belong to any continent. Excluding them from this analysis, `South American` countries owe the most money on average with `240 million`. Although `Africa` appears second on the list of total amount owed, they rank pretty low in average meaning the total amount raked is due to the fact that `Africa` has a lot of coutries when compared to the `Americas`.

# Going into the Debt Purpose of Countries of Interest

## Biggest Debtor - China

In [464]:
China_debt = (debt[debt['Country'] == 'China']
              .groupby(['Indicator_code', 'Indicator_name'])['Debt'].sum().reset_index().sort_values(by='Debt', ascending=False))
China_debt.head(5)

Indicator_code                                     Indicator_name  \
1   DT.AMT.DLXF.CD  Principal repayments on external debt, long-te...   
2   DT.AMT.DPNG.CD  Principal repayments on external debt, private...   
16  DT.INT.DLXF.CD  Interest payments on external debt, long-term ...   
9   DT.DIS.DLXF.CD  Disbursements on external debt, long-term (DIS...   
8   DT.AMT.PRVT.CD          PPG, private creditors (AMT, current US$)   

            Debt  
1   9.621862e+10  
2   7.239299e+10  
16  1.786655e+10  
9   1.569256e+10  
8   1.467746e+10

`China's` biggest indicator is on repayments of long term foreign loan principal at `96 billion`, followed by Debt owed to Private Non-Guaranteed at `72 billion`

In [466]:
China_debt = (debt[debt['Country'] == 'China']
              .groupby(['Indicator_code', 'Indicator_name'])['Debt'].sum().reset_index().sort_values(by='Debt', ascending=True))
China_debt.head(5)

Indicator_code                                   Indicator_name  \
22  DT.INT.PROP.CD  PPG, other private creditors (INT, current US$)   
13  DT.DIS.PROP.CD  PPG, other private creditors (DIS, current US$)   
15  DT.INT.BLAT.CD                PPG, bilateral (INT, current US$)   
7   DT.AMT.PROP.CD  PPG, other private creditors (AMT, current US$)   
18  DT.INT.MLAT.CD             PPG, multilateral (INT, current US$)   

           Debt  
22  156342427.9  
13  334012200.7  
15  514898407.1  
7   796544167.4  
18  858406974.8

## Nigeria

In [494]:
Nigerian_debt = (debt[debt['Country'] == 'Nigeria'])['Debt'].sum()
print(Nigerian_debt)

14149905240.9


The total amount owed by Nigeria is `14 billion`

In [477]:
Nigeria_debt = (debt[debt['Country'] == 'Nigeria']
              .groupby(['Indicator_code', 'Indicator_name'])['Debt'].sum().reset_index().sort_values(by='Debt', ascending=False))
Nigeria_debt.head(10)

Indicator_code                                     Indicator_name  \
1   DT.AMT.DLXF.CD  Principal repayments on external debt, long-te...   
2   DT.AMT.DPNG.CD  Principal repayments on external debt, private...   
10  DT.DIS.OFFT.CD         PPG, official creditors (DIS, current US$)   
8   DT.DIS.DLXF.CD  Disbursements on external debt, long-term (DIS...   
9   DT.DIS.MLAT.CD               PPG, multilateral (DIS, current US$)   
12  DT.INT.DLXF.CD  Interest payments on external debt, long-term ...   
13  DT.INT.DPNG.CD  Interest payments on external debt, private no...   
4   DT.AMT.OFFT.CD         PPG, official creditors (AMT, current US$)   
16  DT.INT.PBND.CD                      PPG, bonds (INT, current US$)   
17  DT.INT.PRVT.CD          PPG, private creditors (INT, current US$)   

            Debt  
1   2.957694e+09  
2   2.573686e+09  
10  1.642511e+09  
8   1.642511e+09  
9   1.444703e+09  
12  1.174954e+09  
13  6.018840e+08  
4   3.840083e+08  
16  3.806220e+08  
17  3.806220e+08

`Nigeria's` top 2 largest debt is owed due to repayment of principal amount on long term external loans and private external loan principal repayment totaling `2.96 billion` and `2.57 billion` respectively

# Debt Domination by Composition 

## Disbursement

In [672]:
disbursement = (debt[debt['Indicator_code'] == 'DT.DIS.DLXF.CD']['Debt'].sum())
print(disbursement)

264701069677.5


The total amount of new loans secured by all the countries is `264 billion`

### By Continent

In [529]:
disbursement = (debt[debt['Indicator_code'] == 'DT.DIS.DLXF.CD']
              .groupby(['Continent'])['Debt'].sum().reset_index().sort_values(by='Debt', ascending=False))
disbursement.head(10)

Continent          Debt
1           Asia  9.985178e+10
6    Unspecified  7.469195e+10
0         Africa  6.840683e+10
5  South America  9.830951e+09
2         Europe  7.532003e+09
3  North America  3.655528e+09
4        Oceania  7.320231e+08

`Asia` incurred the highest amount of external long term debt with `99.8 billion` and `Oceanian countries` have the lowest with `732 million`

### By Country

In [533]:
disbursement = (debt[debt['Indicator_code'] == 'DT.DIS.DLXF.CD']
              .groupby(['Country'])['Debt'].sum().reset_index().sort_values(by='Debt', ascending=False))
disbursement.head(5)

Country          Debt
62   Least developed countries: UN classification  4.016077e+10
51                                            IDA  3.453119e+10
100                                    South Asia  2.930622e+10
20                                       Cameroon  1.818666e+10
23                                          China  1.569256e+10

The `Least Developed Countries according to UN Classification`, though undisclosed, are the biggest debtors under the long term loan disbursement with a total of `40 billion`. If we exlclude ambigous terms such as this, `IDA` and `South Asia`, then `Cameroon` becomes the biggest debtor under new long term loan disbursement with `18 billion`, closely followed by `China` with `15 billion`

In [535]:
disbursement = (debt[debt['Indicator_code'] == 'DT.DIS.DLXF.CD']
              .groupby(['Country', 'Continent'])['Debt'].sum().reset_index().sort_values(by='Debt', ascending=True))
disbursement.head(5)

Country Continent        Debt
98           Somalia    Africa   2456971.6
111            Tonga   Oceania   5529604.8
36           Eritrea    Africa   7203213.6
97   Solomon Islands   Oceania   8645249.9
2            Algeria    Africa  10320772.2

`Somalia` managed to incur the lowest amount under disbursement with just `$2.47 million` owed under long term external loan disbursement

## Interest Payment(Long Term and Private Non Guaranteed)

In [679]:
INT = (debt[debt['Indicator_code'].isin(['DT.INT.DLXF.CD', 'DT.INT.DPNG.CD'])]['Debt'].sum())
print(INT)

300271441098.0


The total amount of new loans secured by all the countries is `300 billion`

### By Continent

In [527]:
INT = (debt[debt['Indicator_code'].isin(['DT.INT.DLXF.CD', 'DT.INT.DPNG.CD'])]
              .groupby(['Continent'])['Debt'].sum().reset_index().sort_values(by='Debt', ascending=False))
INT.head(7)

Continent          Debt
1           Asia  1.472439e+11
5  South America  4.666332e+10
2         Europe  3.791678e+10
3  North America  3.144206e+10
0         Africa  1.963501e+10
6    Unspecified  1.629516e+10
4        Oceania  1.075157e+09

`Asian countries` owe the most money due to interests on long term loans and private non guaranteed loans with `92 billion` and `55 billion` respectively with `America`

### By Country

In [519]:
INT = (debt[debt['Indicator_code'].isin(['DT.INT.DLXF.CD', 'DT.INT.DPNG.CD'])]
              .groupby(['Country'])['Debt'].sum().reset_index().sort_values(by='Debt', ascending=False))
INT.head(5)

Country          Debt
23                China  3.200927e+10
14               Brazil  2.795389e+10
100          South Asia  2.574467e+10
73               Mexico  2.473414e+10
90   Russian Federation  2.149850e+10

`China` owes the highest amount due to interest payment, standing at `32 billion`. `Brazil`, `South Asia`, `Mexico` and `Russia` make up the top 5 with every one of them owing over well over `21 billion`

In [539]:
INT = (debt[debt['Indicator_code'].isin(['DT.INT.DLXF.CD', 'DT.INT.DPNG.CD'])]
              .groupby(['Country', 'Continent'])['Debt'].sum().reset_index().sort_values(by='Debt', ascending=True))
INT.head(5)

Country Continent       Debt
25                    Comoros    Africa   937189.9
109               Timor-Leste      Asia  1042000.0
93      Sao Tome and Principe    Africa  1354432.7
47              Guinea-Bissau    Africa  1462857.5
21   Central African Republic    Africa  2354220.8

`Comoros` owes the least debt due to interest repayment with a figure standing at `937,189`, closely followed by `Timor-Leste` with just a little above a million

### Bilateral Loans

In [685]:
Bilateral = (debt[debt['Indicator_code'].isin(['DT.AMT.BLAT.CD', 'DT.DIS.BLAT.CD', 'DT.INT.BLAT.CD'])]['Debt'].sum())

print(Bilateral)

245173716087.80005


The total amount of new loans secured by all the countries is `245 billion`

### By Continent

In [517]:
Bilateral = (debt[debt['Indicator_code'].isin(['DT.AMT.BLAT.CD', 'DT.DIS.BLAT.CD', 'DT.INT.BLAT.CD'])]
             .groupby('Continent')['Debt'].sum().reset_index().sort_values(by='Debt', ascending=False))

Bilateral.head(10)

Continent          Debt
1           Asia  8.365217e+10
0         Africa  7.960287e+10
6    Unspecified  6.481625e+10
5  South America  7.664453e+09
2         Europe  6.978173e+09
3  North America  2.039321e+09
4        Oceania  4.204872e+08

Standing at `83 billion` in debt, `Asia` owes the most amount due to bilateral loan, followed by `Africa` at `79 billion`. `Oceania` owes the least with just `42 billion` owed in bilateral debt.

### By Country

In [549]:
Bilateral = (debt[debt['Indicator_code'].isin(['DT.AMT.BLAT.CD', 'DT.DIS.BLAT.CD', 'DT.INT.BLAT.CD'])]
              .groupby(['Country', 'Continent'])['Debt'].sum().reset_index().sort_values(by='Debt', ascending=False))
Bilateral.head(5)

Country    Continent          Debt
62  Least developed countries: UN classification  Unspecified  3.930849e+10
20                                      Cameroon       Africa  2.721135e+10
99                                    South Asia         Asia  2.571184e+10
51                                           IDA  Unspecified  2.550776e+10
3                                         Angola       Africa  1.831713e+10

Countries classified by the UN as `Least Developed` owe the most amount due to bilateral loans, with `39 billion`. Ignoring ambiguity, `Cameroon` tops the chart with `27 billion`.

In [556]:
Bilateral = (debt[debt['Indicator_code'].isin(['DT.AMT.BLAT.CD', 'DT.DIS.BLAT.CD', 'DT.INT.BLAT.CD'])]
              .groupby(['Country', 'Continent'])['Debt'].sum().reset_index().sort_values(by='Debt', ascending=True))
Bilateral.head(5)

Country Continent       Debt
47           Guinea-Bissau    Africa    95171.9
93   Sao Tome and Principe    Africa  1326961.8
97         Solomon Islands   Oceania  1491375.0
110                  Tonga   Oceania  9113425.0
98                 Somalia    Africa  9226771.4

`Guinea-Bissau` owes the least with `95,171`

### Bonds

In [692]:
bonds = (debt[debt['Indicator_code'].isin(['DT.AMT.PBND.CD', 'DT.INT.PBND.CD'])]['Debt'].sum())
print(bonds)

130227655373.4


The total amount of new loans secured by all the countries is `130 billion`

### By Continent

In [559]:
bonds = (debt[debt['Indicator_code'].isin(['DT.AMT.PBND.CD', 'DT.INT.PBND.CD'])]
              .groupby(['Continent'])['Debt'].sum().reset_index().sort_values(by='Debt', ascending=False))
bonds.head(10)

Continent          Debt
1           Asia  5.266950e+10
3  North America  3.298362e+10
5  South America  1.926393e+10
0         Africa  1.124169e+10
2         Europe  1.054898e+10
6    Unspecified  3.506678e+09
4        Oceania  1.325000e+07

`Asia` owes the most debt due to bonds at `52 billion`. `Oceania` owes the least with `13 million`

### By Country

In [568]:
bond = (debt[debt['Indicator_code'].isin(['DT.AMT.PBND.CD', 'DT.INT.PBND.CD'])]
              .groupby(['Country', 'Continent'])['Debt'].sum().reset_index().sort_values(by='Debt', ascending=False))
bond.head(5)

Country      Continent          Debt
42     Mexico  North America  2.942027e+10
11      China           Asia  1.105893e+10
64     Turkey           Asia  1.028588e+10
32  Indonesia           Asia  9.346819e+09
12   Colombia  South America  6.760927e+09

`Mexico` owes the most due to bonds at `29 billion`, closest country to `Mexico` is `China` in second place at `11 billion`

In [573]:
bond = (debt[debt['Indicator_code'].isin(['DT.AMT.PBND.CD', 'DT.INT.PBND.CD'])]
              .groupby(['Country', 'Continent'])['Debt'].sum().reset_index().sort_values(by='Debt', ascending=True))
bond.head(5)

Country      Continent       Debt
60                       St. Lucia  North America   285925.9
28                          Guinea         Africa   644000.0
61  St. Vincent and the Grenadines  North America  1792994.1
16                        Dominica  North America  1804071.4
13                           Congo         Africa  8736000.0

`St. Lucia` owes the least amount due to bonds at just `285,925`, followed by `Guinea` at `644,000`.

### Commercial Bank Loans

In [698]:
CBL = (debt[debt['Indicator_code'].isin(['DT.AMT.PCBK.CD', 'DT.DIS.PCBK.CD'])]['Debt'].sum())
print(CBL)

76687539353.80002


The total amount of debt owed due to commercial loans is `76 billion`

### By Continent

In [576]:
CBL = (debt[debt['Indicator_code'].isin(['DT.AMT.PCBK.CD', 'DT.DIS.PCBK.CD'])]
              .groupby(['Continent'])['Debt'].sum().reset_index().sort_values(by='Debt', ascending=False))
CBL.head(10)

Continent          Debt
1           Asia  2.392611e+10
2         Europe  2.329268e+10
0         Africa  9.604562e+09
5  South America  9.098978e+09
6    Unspecified  8.296402e+09
3  North America  2.436130e+09
4        Oceania  3.267268e+07

`Asia` tops the list with `23.9 billion`, closely followd by `Europe` at `23.2 billion`

### By Country

In [582]:
CBL = (debt[debt['Indicator_code'].isin(['DT.AMT.PCBK.CD', 'DT.DIS.PCBK.CD'])]
              .groupby(['Country', 'Continent'])['Debt'].sum().reset_index().sort_values(by='Debt', ascending=False))
CBL.head(5)

Country      Continent          Debt
66                            Russian Federation         Europe  2.207740e+10
17                                         China           Asia  7.823294e+09
12                                        Brazil  South America  7.502985e+09
46  Least developed countries: UN classification    Unspecified  5.257926e+09
70                                    South Asia           Asia  5.224356e+09

`Russia` owes the most amount to commercial banks, with a figure standing at `22 billion`

In [589]:
CBL = (debt[debt['Indicator_code'].isin(['DT.AMT.PCBK.CD', 'DT.DIS.PCBK.CD'])]
              .groupby(['Country', 'Continent'])['Debt'].sum().reset_index().sort_values(by='Debt', ascending=True))
CBL.head(5)

Country      Continent     Debt
28  Eswatini         Africa      0.0
7     Belize  North America      0.0
74  Thailand           Asia      0.0
16      Chad         Africa  11033.6
76     Tonga        Oceania  24603.4

`Eswatini`, `Belize` and `Thailand` all owe nothing to commercial banks, while `Chad` and `Tonga` owe approximately `11,000` and `25,000` respectively

### Multilateral Loans

In [703]:
mutilateral = (debt[debt['Indicator_code'].isin(['DT.AMT.MLAT.CD', 'DT.DIS.MLAT.CD', 'DT.INT.MLAT.CD'])]['Debt'].sum())
print(mutilateral)

178441562614.09998


The total amount of debt owed due to multilateral loans is `178 billion`

### By Continent

In [598]:
mutilateral = (debt[debt['Indicator_code'].isin(['DT.AMT.MLAT.CD', 'DT.DIS.MLAT.CD', 'DT.INT.MLAT.CD'])]
              .groupby(['Continent'])['Debt'].sum().reset_index().sort_values(by='Debt', ascending=False))
mutilateral.head(10)

Continent          Debt
1           Asia  7.462185e+10
6    Unspecified  4.243587e+10
0         Africa  3.032655e+10
5  South America  1.389015e+10
2         Europe  1.004357e+10
3  North America  6.551342e+09
4        Oceania  5.722280e+08

`Asia` holds the largest amount of multilateral debt, with a figure standing at `74 billion`. `Oceania` owes the lowest amount, standing at `572 million`

### By Country

In [603]:
mutilateral = (debt[debt['Indicator_code'].isin(['DT.AMT.MLAT.CD', 'DT.DIS.MLAT.CD', 'DT.INT.MLAT.CD'])]
              .groupby(['Country', 'Continent'])['Debt'].sum().reset_index().sort_values(by='Debt', ascending=False))
mutilateral.head(5)

Country    Continent          Debt
51                                            IDA  Unspecified  2.301490e+10
100                                    South Asia         Asia  2.148996e+10
62   Least developed countries: UN classification  Unspecified  1.942096e+10
52                                          India         Asia  1.074128e+10
23                                          China         Asia  6.553632e+09

Ambiguity aside, `India` owes the most amount of multilateral debt, with a figure amounting to `23 billion`

In [606]:
mutilateral = (debt[debt['Indicator_code'].isin(['DT.AMT.MLAT.CD', 'DT.DIS.MLAT.CD', 'DT.INT.MLAT.CD'])]
              .groupby(['Country', 'Continent'])['Debt'].sum().reset_index().sort_values(by='Debt', ascending=True))
mutilateral.head(5)

Country Continent        Debt
25                 Comoros    Africa   7303720.5
114           Turkmenistan      Asia   7980189.2
2                  Algeria    Africa   8094779.0
111                  Tonga   Oceania   9265073.2
93   Sao Tome and Principe    Africa  13605715.7

`Comoros` holds the least amount of multilateral debt, at `7.3 million`

### Official Creditors

In [706]:
OFC = (debt[debt['Indicator_code'].isin(['DT.AMT.OFFT.CD', 'DT.DIS.OFFT.CD', 'DT.INT.OFFT.CD'])]['Debt'].sum())
print(OFC)

423615278701.9


The total amount of debt owed due to Official Creditors loans is `423 billion`

### By Continent

In [613]:
OFC = (debt[debt['Indicator_code'].isin(['DT.AMT.OFFT.CD', 'DT.DIS.OFFT.CD', 'DT.INT.OFFT.CD'])]
              .groupby(['Continent'])['Debt'].sum().reset_index().sort_values(by='Debt', ascending=False))
OFC.head(10)

Continent          Debt
1           Asia  1.582740e+11
0         Africa  1.099294e+11
6    Unspecified  1.072521e+11
5  South America  2.155461e+10
2         Europe  1.702175e+10
3  North America  8.590663e+09
4        Oceania  9.927152e+08

`Asia` owes the most amount of offcial creditors debt at `158 billion`. `Oceania` owes the least, at `992 million`

### By Country

In [623]:
OFC = (debt[debt['Indicator_code'].isin(['DT.AMT.OFFT.CD', 'DT.DIS.OFFT.CD', 'DT.INT.OFFT.CD'])]
              .groupby(['Country', 'Continent'])['Debt'].sum().reset_index().sort_values(by='Debt', ascending=False))
OFC.head(5)

Country    Continent          Debt
62   Least developed countries: UN classification  Unspecified  5.872945e+10
51                                            IDA  Unspecified  4.852266e+10
100                                    South Asia         Asia  4.720179e+10
20                                       Cameroon       Africa  2.825860e+10
34                                          Egypt       Africa  1.951404e+10

`Cameroon` owes the largest amount of Official Creditors debt, at `28 billion`, followed by `Egypt` at `19 billion`

In [625]:
OFC = (debt[debt['Indicator_code'].isin(['DT.AMT.OFFT.CD', 'DT.DIS.OFFT.CD', 'DT.INT.OFFT.CD'])]
              .groupby(['Country', 'Continent'])['Debt'].sum().reset_index().sort_values(by='Debt', ascending=True))
OFC.head(5)

Country Continent        Debt
93   Sao Tome and Principe    Africa  14932677.5
97         Solomon Islands   Oceania  15494489.1
111                  Tonga   Oceania  18378498.2
25                 Comoros    Africa  19189494.4
109            Timor-Leste      Asia  23666986.4

`Sao Tome and Principe` owes the least amount of Official Creditors debt, at `149 million`.

### Private Creditors

In [710]:
PVC = (debt[debt['Indicator_code'].isin(['DT.AMT.PRVT.CD', 'DT.DIS.PRVT.CD', 'DT.INT.PRVT.CD'])]['Debt'].sum())
print(PVC)

263796692669.7


The total amount of debt owed due to multilateral loans is `263 billion`

### By Continent

In [629]:
PVC = (debt[debt['Indicator_code'].isin(['DT.AMT.PRVT.CD', 'DT.DIS.PRVT.CD', 'DT.INT.PRVT.CD'])]
              .groupby(['Continent'])['Debt'].sum().reset_index().sort_values(by='Debt', ascending=False))
PVC.head(10)

Continent          Debt
1           Asia  8.315658e+10
5  South America  6.648863e+10
2         Europe  3.850042e+10
3  North America  3.644645e+10
0         Africa  2.381586e+10
6    Unspecified  1.531828e+10
4        Oceania  7.046728e+07

`Asia` owes the most amount due to private creditors debt at `83 billion`. `Oceania` owe the least, at `70 million`

### By Country

In [631]:
PVC = (debt[debt['Indicator_code'].isin(['DT.AMT.PRVT.CD', 'DT.DIS.PRVT.CD', 'DT.INT.PRVT.CD'])]
              .groupby(['Country', 'Continent'])['Debt'].sum().reset_index().sort_values(by='Debt', ascending=False))
PVC.head(5)

Country      Continent          Debt
13              Brazil  South America  4.957300e+10
62              Mexico  North America  3.176416e+10
77  Russian Federation         Europe  2.991993e+10
19               China           Asia  2.113905e+10
46           Indonesia           Asia  1.262385e+10

`Brazil` owes the most amount at `49 billion`, followed by `Mexico` at `31 billion`

In [633]:
PVC = (debt[debt['Indicator_code'].isin(['DT.AMT.PRVT.CD', 'DT.DIS.PRVT.CD', 'DT.INT.PRVT.CD'])]
              .groupby(['Country', 'Continent'])['Debt'].sum().reset_index().sort_values(by='Debt', ascending=True))
PVC.head(5)

Country Continent      Debt
89     Tonga   Oceania   24603.4
60      Mali    Africa   68120.2
25  Djibouti    Africa  138000.0
31   Eritrea    Africa  170300.6
56   Lesotho    Africa  237679.5

`Tonga` owes the least amount at `24,000`

### Other Private Creditors

In [713]:
OPC = (debt[debt['Indicator_code'].isin(['DT.AMT.PROP.CD', 'DT.DIS.PROP.CD', 'DT.INT.PROP.CD'])]['Debt'].sum())
print(OPC)

43723098445.799995


The total amount of debt owed due to Official Private Creditor loans is `43 billion`

### By Continent

In [640]:
OPC = (debt[debt['Indicator_code'].isin(['DT.AMT.PROP.CD', 'DT.DIS.PROP.CD', 'DT.INT.PROP.CD'])]
              .groupby(['Continent'])['Debt'].sum().reset_index().sort_values(by='Debt', ascending=False))
OPC.head(10)

Continent          Debt
4  South America  3.588607e+10
1           Asia  3.640828e+09
5    Unspecified  2.165012e+09
0         Africa  1.552919e+09
2         Europe  3.185704e+08
3  North America  1.596963e+08

By a large margin, `South America` owes the most debt amount to Other Private Creditors at `35 billion`. In second place is `Asia` at `3.6 billion`. `North America` owes the least at `159 million`

### By Country

In [644]:
OPC = (debt[debt['Indicator_code'].isin(['DT.AMT.PROP.CD', 'DT.DIS.PROP.CD', 'DT.INT.PROP.CD'])]
              .groupby(['Country', 'Continent'])['Debt'].sum().reset_index().sort_values(by='Debt', ascending=False))
OPC.head(5)

Country      Continent          Debt
8                                         Brazil  South America  3.573242e+10
26                                     Indonesia           Asia  1.753962e+09
12                                         China           Asia  1.286899e+09
31  Least developed countries: UN classification    Unspecified  1.224848e+09
24                                           IDA    Unspecified  9.401638e+08

`Brazil` owes the most at `35 billion`, `Indonesia` owes the second most with `175 million`

In [646]:
OPC = (debt[debt['Indicator_code'].isin(['DT.AMT.PROP.CD', 'DT.DIS.PROP.CD', 'DT.INT.PROP.CD'])]
              .groupby(['Country', 'Continent'])['Debt'].sum().reset_index().sort_values(by='Debt', ascending=True))
OPC.head(5)

Country      Continent      Debt
1     Algeria         Africa   23129.8
49   Tanzania         Africa   45783.8
33  Macedonia         Europe   66332.2
23   Honduras  North America   73052.6
19    Eritrea         Africa  170300.6

`Algeria` owes the most with `23,000`. `Tanzania`, `Macedonia`, `Honduras` and `Eritea` make up the bottom 5, with `Eritea` owing `170,000`.

### Principal Repayment

In [717]:
PRP = (debt[debt['Indicator_code'].isin(['DT.AMT.DLXF.CD', 'DT.AMT.DPNG.CD'])]['Debt'].sum())
print(PRP)

1139938034157.1


The total amount of debt owed due to Principal Repayment is `1 trillion`

### By Continent

In [653]:
PRP = (debt[debt['Indicator_code'].isin(['DT.AMT.DLXF.CD', 'DT.AMT.DPNG.CD'])]
              .groupby(['Continent'])['Debt'].sum().reset_index().sort_values(by='Debt', ascending=False))
PRP.head(10)

Continent          Debt
1           Asia  5.884953e+11
5  South America  1.854162e+11
2         Europe  1.711666e+11
0         Africa  8.397212e+10
6    Unspecified  5.580122e+10
3  North America  4.936843e+10
4        Oceania  5.718231e+09

`Asia` owes the most amount due to Amortization, standing at `588 billion`, `Oceania` owes the least at `5.7 billion`

### By Country

In [660]:
PRP = (debt[debt['Indicator_code'].isin(['DT.AMT.DLXF.CD', 'DT.AMT.DPNG.CD'])]
              .groupby(['Country', 'Continent'])['Debt'].sum().reset_index().sort_values(by='Debt', ascending=False))
PRP.head(5)

Country      Continent          Debt
23                China           Asia  1.686116e+11
14               Brazil  South America  1.318733e+11
90   Russian Federation         Europe  1.093899e+11
113              Turkey           Asia  9.147123e+10
100          South Asia           Asia  7.312715e+10

`China` owes the most amount with an `168 billion` debt due to amortization, followed by `Brazil` at `131 billion`

In [658]:
PRP = (debt[debt['Indicator_code'].isin(['DT.AMT.DLXF.CD', 'DT.AMT.DPNG.CD'])]
              .groupby(['Country', 'Continent'])['Debt'].sum().reset_index().sort_values(by='Debt', ascending=True))
PRP.head(5)

Country Continent        Debt
109            Timor-Leste      Asia    825000.0
93   Sao Tome and Principe    Africa   2941396.3
25                 Comoros    Africa   4792269.1
47           Guinea-Bissau    Africa   6783629.1
111                  Tonga   Oceania  10369910.6

`Timor-Leste` owes the least amount at `825,000`

# INSIGHTS
- `Nigeria’s` burden comes largely from repaying older loans, not new disbursements.
- `China` is the world's biggest debtor with `285.8 billion`, largely due to principal repayments on long-term foreign loans `96 billion` and private non-guaranteed debt `72 billion`, `Brazil` is a close second with `280.6 billion`.
- `China` and `Brazil's` dominance in the debt figures reflects large-scale international borrowing, possibly linked to massive infrastructure projects and energy investments.
- Other major debtors include `South Asia region`, `UN’s Least Developed Countries`, and `Russia`.
- `Sao Tome and Principe` has the lowest total debt: `44.8 million`.
- Other low-debt countries include `Tonga`, `Comoros`, `Timor-Leste`. These countries are generally small economies with limited borrowing capacity or better debt discipline.
- `Asia` dominates the global external debt while `Oceania` does not borrow as much as others.
- `South America` has the highest average debt per country, approximately `2.4 billion`.
- `Africa` has a lower average, `508 million`, despite being 2nd in total showing high country count but smaller individual loans.
- `South American` countries owe the most money per country with `240 million`

# STRATEGIC INSIGHTS
- `China` is structurally indebted across nearly every loan category. This implies a deliberate economic strategy built on external financing.
- `South America`, while not highest in total, has the most per-country debt which suggests deeper borrowing per country than other continents.
- `Africa` appears heavily exposed to official and bilateral debt, which may tie it closely to conditional financing programs.
- `Oceania's` consistent position at the bottom across every debt type signals either strong fiscal discipline or limited access to capital markets.
- Multilateral loans (World Bank, IMF) play a strong stabilizing role, especially in `Africa` and `Asia`.